modules/core/shared/src/main/scala-3/io/circe/Derivation.scala

```diff
@@ -1,300 +1,52 @@
+import Predef.genericArrayOps
+import cats.data.{ NonEmptyList, Validated }
+import io.circe.derivation._
 
+@deprecated(since = "0.14.4")
 object Derivation {
-
   inline final def summonLabels[T <: Tuple]: Array[String] = summonLabelsRec[T].toArray
-  inline final def summonDecoders[T <: Tuple]: Array[Decoder[_]] = summonDecodersRec[T].toArray
-  inline final def summonEncoders[T <: Tuple]: Array[Encoder[_]] = summonEncodersRec[T].toArray
-
-  inline final def summonEncoder[A]: Encoder[A] = summonFrom {
-    case encodeA: Encoder[A] => encodeA
-    case _: Mirror.Of[A]     => Encoder.AsObject.derived[A]
-  }
-
-  inline final def summonDecoder[A]: Decoder[A] = summonFrom {
-    case decodeA: Decoder[A] => decodeA
-    case _: Mirror.Of[A]     => Decoder.derived[A]
-  }
+  inline final def summonDecoders[T <: Tuple]: Array[Decoder[_]] = summonDecoders[T].toArray
+  inline final def summonEncoders[T <: Tuple]: Array[Encoder[_]] = summonEncoders[T].toArray
 
-  inline final def summonLabelsRec[T <: Tuple]: List[String] = inline erasedValue[T] match {
-    case _: EmptyTuple => Nil
-    case _: (t *: ts)  => constValue[t].asInstanceOf[String] :: summonLabelsRec[ts]
-  }
+  inline final def summonEncoder[A]: Encoder[A] = derivation.summonEncoder[A](using Configuration.default)
+  inline final def summonDecoder[A]: Decoder[A] = derivation.summonDecoder[A](using Configuration.default)
 
+  inline final def summonLabelsRec[T <: Tuple]: List[String] = derivation.summonLabels[T]
   inline final def summonDecodersRec[T <: Tuple]: List[Decoder[_]] =
-    inline erasedValue[T] match {
-      case _: EmptyTuple => Nil
-      case _: (t *: ts)  => summonDecoder[t] :: summonDecodersRec[ts]
-    }
-
+    derivation.summonDecoders[T](using Configuration.default)
   inline final def summonEncodersRec[T <: Tuple]: List[Encoder[_]] =
-    inline erasedValue[T] match {
-      case _: EmptyTuple => Nil
-      case _: (t *: ts)  => summonEncoder[t] :: summonEncodersRec[ts]
-    }
+    derivation.summonEncoders[T](using Configuration.default)
 }
 
-private[circe] trait EncoderDerivation {
-  inline final def derived[A](using inline A: Mirror.Of[A]): Encoder.AsObject[A] =
-    new DerivedEncoder[A]
-      with DerivedInstance[A](
-        constValue[A.MirroredLabel],
-        Derivation.summonLabels[A.MirroredElemLabels]
-      ) {
-      protected[this] lazy val elemEncoders: Array[Encoder[_]] =
-        Derivation.summonEncoders[A.MirroredElemTypes]
-
-      final def encodeObject(a: A): JsonObject = inline A match {
-        case m: Mirror.ProductOf[A] =>
-          JsonObject.fromIterable(encodedIterable(a.asInstanceOf[Product]))
-        case m: Mirror.SumOf[A] =>
-          encodeWith(m.ordinal(a))(a) match {
-            case (k, v) => JsonObject.singleton(k, v)
-          }
-      }
-    }
-}
+@deprecated(since = "0.14.4")
+private[circe] trait DerivedInstance
+private[circe] trait DerivedEncoder
+private[circe] trait DerivedDecoder
```

Comment:
Those were being used in:
https://github.com/circe/circe/blob/series/0.14.x/modules/core/shared/src/main/scala-3/io/circe/Derivation.scala#L44
https://github.com/circe/circe/blob/series/0.14.x/modules/core/shared/src/main/scala-3/io/circe/Derivation.scala#L65
https://github.com/circe/circe/blob/series/0.14.x/modules/core/shared/src/main/scala-3/io/circe/Derivation.scala#L137
But they were already `private[circe]` so there is no need to mark them deprecated.

Reply:
its my understanding that another artifact within the circe org could still reference those traits since they will be within the `io.circe` package.  I would like them marked deprecated to prevent them from accidentally being referenced elsewhere.

Reply:
Go ahead and mark these deprecated for me and assuming the build passes I think we can get this merged in and a release cut this week. 

Thank you for your work on this, its greatly appreciated!